lib/jobs/local-catalog.js

```diff
@@ -0,0 +1,180 @@
+// Copyright 2015, EMC, Inc.
+/* jshint: node:true */
+
+'use strict';
+
+var di = require('di');
+
+module.exports = LocalCatalogJobFactory;
+di.annotate(LocalCatalogJobFactory, new di.Provide('Job.Local.Catalog'));
+    di.annotate(LocalCatalogJobFactory,
+    new di.Inject(
+        'Job.Base',
+        'JobUtils.CommandParser',
+        'Services.Waterline',
+        'Logger',
+        'Promise',
+        'Assert',
+        'Util',
+        '_',
+        'Services.Encryption',
+        'ChildProcess'
+    )
+);
+function LocalCatalogJobFactory(
+    BaseJob, 
+    parser, 
+    waterline, 
+    Logger, 
+    Promise, 
+    assert, 
+    util, 
+    _,
+    encryption,
+    ChildProcess
+) {
+    var logger = Logger.initialize(LocalCatalogJobFactory);
+
+    /**
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function LocalCatalogJob(options, context, taskId) {
+        LocalCatalogJob.super_.call(this, 
+                                   logger,
+                                   options, 
+                                   context, 
+                                   taskId);
+
+        assert.string(this.context.target);
+        assert.arrayOfString(this.options.commands);
+
+        this.nodeId = this.context.target;
+        this.commands = _.isArray(options.commands) ? options.commands : [options.commands];
+    }
+    util.inherits(LocalCatalogJob, BaseJob);
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype._run = function() {
+        var self = this;
+
+        waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            assert.ok(node, 'No node for local catalog');
+            return Promise.map(self.commands, function(cmd) {
+                return parser.validateParser(cmd)
+                .then(function() {
+                    return self.runCommand(cmd);
+                })
+                .catch(function(err) {
+                    logger.error("Run error", { command: cmd, error: err });
+                });
+            });
+        })
+        .then(function(cmdResp) {
+            return cmdResp;
+        })
+        .then(self.handleResponse.bind(self))
+        .then(function() {
+            self._done();
+        })
+        .catch(function(err) {
+            self._done(err);
+        });
+            
+    };
+
+    /**
+     * @memberOf LocalCatalogJob
+     */
+    LocalCatalogJob.prototype.runCommand = function (cmd) {
+        var self = this;
+        var args = cmd.split(' ');
+        var program = args[0];
+        if(program === "sudo") {
+            program = args[1];
+            args.shift();
+        }
+        args.shift();
+        return Promise.resolve()
+        .then(function() {
+            var env = {
+                'nodeid': self.nodeId
+            };
+            var childProcess = new ChildProcess(program, args, env);
```

Comment:
One thing that would be good to support in this job is adding a _cleanup() function to the job that kills this ChildProcess, that way if someone cancels the workflow we don't have a bunch of background child processes still running. It's probably not critical since the catalog commands shouldn't change state, but it's still nice to know we're not leaking processes for any amount of time.